R/plot_taxa.R

```diff
@@ -422,3 +758,548 @@ format_for_comm_plots <- function(observation, id) {
+  }
+  
+
+  validate_arguments(fun.name = "plot", fun.args = as.list(environment()))
+  
+  p <- data_long %>%
+    ggplot2::ggplot(
+      ggplot2::aes(taxon_rank)) + 
+    ggplot2::labs(title = "Taxa rank frequencies in the observation table", subtitle = id) +
+    ggplot2::xlab("Taxon rank") +
+    ggplot2::ylab(paste0("Number of occurrences")) +
+    ggplot2::geom_bar() +
+    ggplot2::theme_bw() +
+    ggplot2::coord_flip() 
+    # ggplot2::theme(
+    #   axis.text.x = ggplot2::element_text(angle = 45, hjust=1))
+  
+  if(!is.na(facet_var)){
+    p <- p +
+      ggplot2::facet_wrap(
+        ~ as.factor(.data[[facet_var]]),
+        scales = facet_scales)
+  }
+  
+  return(p)
+}
+
+
+
+
+
+
+
+
+
+
+#' Plot stacked taxa by site
+#'
+#' @param dataset (list) An ecocomDP formatted dataset
+#' @param flat_data (tbl_df, tbl, data.frame) A flat data.frame created from an \code{ecocomDP} formatted dataset using \code{ecocomDP::flatten_data}.
+#' @param id (character) Identifier of dataset to be used in plot subtitles.
+#' @param rank (string) The rank of taxa to plot. Defaults to NA, which will inlcude all ranks.
+#' @param cutoff (numeric) Minimum number of occurrences allowed for taxa included in the plot.
+#' @param facet_var (character) Name of column to use for faceting.
+#' @param facet_scales (character) Should scales be free ("free", default value), fixed ("fixed"), or free in one dimension ("free_x", "free_y")?
+#' @param color_var (character) Name of column to use for plot colors.
+#' @param alpha (numeric) Alpha-transparency scale of data points. Useful when many data points overlap. Allowed values are between 0 and 1, where 1 is 100\% opaque. Default is 1.
+#' 
+#' @return (gg, ggplot) A gg, ggplot object if assigned to a variable, otherwise a plot to your active graphics device.
+#' 
+#' @import dplyr
+#' @import ggplot2
+#' @import tidyr
+#' 
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' # plot ecocomDP formatted dataset
+#' plot_taxa_occurrence_frequency(
+#'   dataset = ants_L1)
+#' 
+#' # plot flattened ecocomDP dataset
+#' plot_taxa_occurrence_frequency(
+#'   flat_data = flatten_dataset(ants_L1))
+#' 
+#' # facet by location color by taxon_rank
+#' plot_taxa_occurrence_frequency(
+#'   dataset = ants_L1,
+#'   facet_var = "location_id",
+#'   color_var = "taxon_rank")
+#' 
+#' # facet by location, only plot taxa of rank = "species"
+#' plot_taxa_occurrence_frequency(
+#'   dataset = ants_L1,
+#'   facet_var = "location_id",
+#'   rank = "Species", cutoff = 2)
+#' 
+#' # color by location, only include taxa with > 10 occurrences
+#' plot_taxa_occurrence_frequency(
+#'   dataset = ants_L1,
+#'   color_var = "location_id",
+#'   cutoff = 5)
+#' 
+#' # tidy syntax, filter data by date
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     lubridate::as_date(datetime) > "2003-07-01") %>%
+#'   plot_taxa_occurrence_frequency(flat_data = .)
+#' }
+plot_taxa_occurrence_frequency <- function(
```

Comment:
Same comments as `plot_taxa_accum_sites()`

Reply:
this guidance will not work for this plot since it was originally written to expect a fully flattened dataset.

Reply:
Use the argument subsuming approach offered in plot_taxa_accum_sites(), but only allow the `flat_data` type?